cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -211,13 +215,25 @@ public BodyConsumer deploy(HttpRequest request, HttpResponder responder,
                              @HeaderParam(ARCHIVE_NAME_HEADER) final String archiveName) {
     // null means use name provided by app spec
     try {
-      return deployApplication(responder, namespaceId, null, archiveName);
+      boolean suspendSchedules = suspendScheduleOnDeploy(request);
+      return deployApplication(responder, namespaceId, null, archiveName, suspendSchedules);
     } catch (Exception ex) {
       responder.sendString(HttpResponseStatus.INTERNAL_SERVER_ERROR, "Deploy failed: " + ex.getMessage());
       return null;
     }
   }
 
+  private boolean suspendScheduleOnDeploy(HttpRequest request) throws URISyntaxException {
+    // Optional argument to suspend schedules upon application deployment
+    Map<String, List<String>> queryParams = new QueryStringDecoder(new URI(request.getUri())).getParameters();
+    boolean suspendSchedules = false;
+    List<String> values = queryParams.get(Constants.Scheduler.SUSPEND_SCHEDULES_DEPLOY_OPTION);
```

Comment:
I think we can use `@QueryParam` instead of manually parsing the query param here.